platform/measurement_utils.cpp

```diff
@@ -18,12 +18,28 @@
 
 namespace measurement_utils
 {
-std::string ToStringPrecision(double d, int pr)
+// Global Locale variable to store system locale.
+static platform::Locale g_systemLocale = {
+        "",  // Undefined language.
+        "",  // Undefined country.
+        "",  // Undefined currency.
+        ".", // Dot as default decimal separator.
+        ","  // Comma as default grouping (thousands) separator.
+};
+
+void SetSystemLocale(platform::Locale const & locale)
 {
-  // We assume that the app will be restarted if a user changes device's locale.
-  static auto const locale = platform::GetCurrentLocale();
```

Comment:
The previous approach initialized locale on startup without any additional calls/refreshes. Now what happens on iOS or Linux or Mac?

Reply:
An unnecessary condition check on every call is not a good solution.

What's wrong with the old approach of initializing the global variable, but making it non-const, and updating it from the UI only when necessary/when it really changed?

Reply:
> What's wrong with the old approach of initializing the global variable, but making it non-const, and updating it from the UI only when necessary/when it really changed?

I have no problem with keeping the global variable. Right now we're initializing it on both iOS and Android.

For Linux/macOS desktop, further checks shall be made on a separate issue after this PR is merged.

Reply:
> Now what happens on iOS or Linux or Mac?

I've also implemented the locale initialization at app start-up on Linux.